src/java.desktop/windows/native/libawt/windows/awt_Win32GraphicsDevice.cpp

```diff
@@ -137,7 +137,13 @@ HDC AwtWin32GraphicsDevice::MakeDCFromMonitor(HMONITOR hmMonitor) {
             HDC hDC = CreateDC(mieInfo.szDevice, NULL, NULL, NULL);
             if (NULL != hDC) {
                 retCode = hDC;
+            } else {
+                J2dTraceLn1(J2D_TRACE_WARNING,
+                       "CreateDC failed, so we cannot store the DC for later usage, mieInfo.szDevice is %S",
```

Comment:
Is `%S` a valid format string? It should in lower case: `%s`.

It seems, “store the DC for later usage” is misleading, it isn't stored… at least for a long period of time.

Reply:
https://learn.microsoft.com/en-us/cpp/c-runtime-library/format-specification-syntax-printf-and-wprintf-functions?view=msvc-170
The %S is for wide chars, do you think we need the other one (%s) ?

Reply:
> https://learn.microsoft.com/en-us/cpp/c-runtime-library/format-specification-syntax-printf-and-wprintf-functions?view=msvc-170 The %S is for wide chars, do you think we need the other one (%s) ?

You're right, it should be wide-char.